doc/releases/changelog-dev.md

```diff
@@ -130,13 +130,19 @@
 
 <h3>Improvements 🛠</h3>
 
+* When using `defer_measurements` with postselecting mid-circuit measurements, operations
+  that will never be active due to the postselected state are skipped in the transformed
+  quantum circuit. In addition, postselected controls are skipped, as they are evaluated
+  at transform time.
```

Comment:
```suggestion
  at transform time. This optimization can be turned off with the `reduce_postselected` keyword argument.
```

Reply:
Optional, but examples are always appreciated!

Reply:
Added mention of switch and example locally.